### PR TITLE
[image_picker] Fix 'messages.g.h' file not found

### PR DESCRIPTION
**Pull request by hueyzng**
Screen Shot 2022-05-05 at 14 53 32

Fixes: https://github.com/flutter/flutter/issues/103629

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Thanks for the contribution! You’ve checked boxes in the PR checklist above that are not reflected in this PR, so I’m assuming this is a work in progress and am marking it as a Draft. Please review the checklist, updating the PR as appropriate, and when the checklist reflects the state of the PR as posted please feel free to mark it as ready for review.

In particular, you haven't done either of these steps, which are required:

> * [x]  I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates).
> * [x]  I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style).

File: packages/image_picker/image_picker/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.8.6
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/CHANGELOG.md:1]: Please revert the changes to `image_picker`; there's no need to update the app-facing dependencies for a bugfix in an implementation package.

File: packages/image_picker/image_picker_ios/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 * Removes unnecessary imports.
 
+## 0.8.6
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_ios/CHANGELOG.md:5]: Please see https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#updating-a-changelog-that-has-a-next

File: packages/image_picker/image_picker_ios/pubspec.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: image_picker_ios
 description: iOS implementation of the video_picker plugin.
 repository: https://github.com/flutter/plugins/tree/main/packages/image_picker/image_picker_ios
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+image_picker%22
-version: 0.8.5
+version: 0.8.6
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_ios/pubspec.yaml:5]: This is a bugfix of a pre-1.0 package, so should update the build number instead. So this should be 0.8.5+1.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@hueyzng By the way, is there anything unusual about your setup (e.g., are you using add-to-app or something along those lines)? The change is clearly correct, but I'm wondering why our CI wasn't affected by the issue.

**Comment by hueyzng** [comment]
@stuartmorgan I don't have any special settings. I just run flutter pub add image_picker and then run the ios project and it reports this error.

**Comment by hueyzng** [comment]
Please review my changes again is there anything else wrong with them. I'd be happy to fix it.

**Comment by stuartmorgan** [comment]
> I don't see CI running build tests on a mac

I'm not sure where you are looking exactly, but every plugin is built for every supported platform (multiple times, in fact) on every CI run.

**Comment by hueyzng** [comment]
Sorry it's my fault. I only just saw the .ci/scripts/build_all_plugins.sh file. Now I see the macOS tasks at .cirrus.yml. When I try to build packages/image_picker/image_picker_ios/example locally, it does build successfully using both angled and quotes include. I don't understand how using angled include can be built successfully, but on my project that is failed.

image

**Comment by stuartmorgan** [comment]
> I don't understand how using angled include can be built successfully, but failed in my project.

It may depend on project-level search path settings.

File: packages/image_picker/image_picker/pubspec.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@ dependencies:
     sdk: flutter
   image_picker_android: ^0.8.4+11
   image_picker_for_web: ^2.1.0
-  image_picker_ios: ^0.8.4+11
+  image_picker_ios: ^0.8.5+1
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/pubspec.yaml:27]: Please revert this change, per my previous comment.

File: packages/image_picker/image_picker_ios/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
-## NEXT
+## 0.8.5+1
 
 * Removes unnecessary imports.
+* Fix 'messages.g.h' file not found.
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_ios/CHANGELOG.md:4]: Fixes

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by Hixie** [comment]
i presume that testing this is intractable because of the number of potential configurations we would have to test to catch this kind of thing?

test-exempt: intractable, assuming i'm right that it would not be practical to actually test for this

**Comment by stuartmorgan** [comment]
> i presume that testing this is intractable because of the number of potential configurations we would have to test to catch this kind of thing?

I checked the example, and I don't see anything that would be causing us to find local headers when using system include style (`<>`). If we figure out what causes it we could change it in our example apps, but it looks like it's more subtle than just "remove some Xcode-default-template set of search paths".

@hueyzng If you see what's different in your project's build settings and the example's build settings that causes this, please let us know.

File: packages/image_picker/image_picker_ios/CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ## 0.8.5+1
 
 * Removes unnecessary imports.
+* Fixes 'messages.g.h' file not found.
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_ios/CHANGELOG.md:4]: You've merged incorrectly here; you're now adding to the change list for a version that has already been published. You need a new version bump.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by keviner2004** [comment]
I have the error when I use `flutter run` to run `module` project with dependency `image_picker ^0.8.5+1` on iOS simulator.

`Lexical or Preprocessor Issue (Xcode): 'messages.g.h' file not found with <angled> include; use "quotes" instead`

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
@cyanglaz Ping on final review here.

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks!

**Comment by mukkoticf** [comment]
Please update this fix in **image_picker** package also. I can see **image_picker_ios** version is old in pubspec.yaml of **image_picker**

**Comment by stuartmorgan** [comment]
> Please update this fix in **image_picker** package also.

This issue doesn't exist in that package, so there's nothing to fix.

> I can see **image_picker_ios** version is old in pubspec.yaml of **image_picker**

No, you can see that the *minimum* version is older than this. Which is an accurate expression of what that package actually requires.

You just need to `dart pub upgrade`.

**Comment by mukkoticf** [comment]
Thank you @stuartmorgan . This worked

**Pull request merged**